src/main/java/seedu/intern/logic/commands/EditAllCommand.java

```diff
@@ -0,0 +1,171 @@
+        for (Applicant applicantToEdit : lastShownList) {
+            Applicant editedApplicant = createEditedApplicant(applicantToEdit, editAllDescriptor);
+            model.setApplicant(applicantToEdit, editedApplicant);
+            addSuccesses++;
+        }
+
+        String result = addSuccesses + " of " + lastShownList.size() + " applicants updated\n";
+
+        model.updateFilteredApplicantList(PREDICATE_SHOW_ALL_PERSONS);
+        return new CommandResult(String.format(MESSAGE_EDIT_ALL_SUCCESS, result));
+    }
+
+    /**
+     * Creates and returns a {@code Applicant} with the details of {@code applicantToEdit}
+     * edited with {@code EditAllDescriptor}.
+     */
+    private static Applicant createEditedApplicant(Applicant applicantToEdit,
+                                                EditAllDescriptor editAllDescriptor) {
+        assert applicantToEdit != null;
+
+        Name updatedName = applicantToEdit.getName();
+        Phone updatedPhone = applicantToEdit.getPhone();
+        Email updatedEmail = applicantToEdit.getEmail();
+        Grade updatedGrade = applicantToEdit.getGrade();
+        Institution updatedInstitution = applicantToEdit.getInstitution();
+        GraduationYearMonth updatedGraduationYearMonth = applicantToEdit.getGraduationYearMonth();
+        Course updatedCourse = applicantToEdit.getCourse();
+        Set<Skill> updatedSkills = applicantToEdit.getSkills();
+
+        ApplicationStatus updatedStatus = editAllDescriptor.getApplicationStatus()
+                .orElse(applicantToEdit.getApplicationStatus());
+
+        return new Applicant(updatedName, updatedPhone, updatedEmail,
+                updatedGrade, updatedInstitution, updatedCourse,
+                updatedGraduationYearMonth, updatedStatus, updatedSkills);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof EditAllCommand)) {
+            return false;
+        }
+
+        // state check
+        EditAllCommand e = (EditAllCommand) other;
+        return editAllDescriptor.equals(e.editAllDescriptor);
+    }
+
+    public String toString() {
+        return editAllDescriptor.toString();
+    }
+
+    /**
+     * Stores the details to update all applicants with. Each non-empty field value will replace the
+     * corresponding field value of all currently filtered applicants.
+     */
+    public static class EditAllDescriptor {
+        private ApplicationStatus status;
+
+        public EditAllDescriptor() {}
+
+        /**
+         * Copy constructor.
+         */
+        public EditAllDescriptor(EditAllDescriptor toCopy) {
+            setApplicationStatus(toCopy.status);
+        }
+
+        /**
+         * Returns true if at least one field is edited.
+         */
+        public boolean isAnyFieldEdited() {
```

Comment:
I think the only possible detail for mass changing will be the status only but we may still keep this to allow room for modifications/update later on